google-cloud-resourcemanager/src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java

```diff
@@ -16,58 +16,198 @@
     resourceManager =
-        new Cloudresourcemanager.Builder(transport, new JacksonFactory(), initializer)
+        new CloudResourceManager.Builder(transport, new JacksonFactory(), initializer)
             .setRootUrl(options.getHost())
             .setApplicationName(options.getApplicationName())
             .build();
+    clock = options.getClock();
   }
 
   private static ResourceManagerException translate(IOException exception) {
     return new ResourceManagerException(exception);
   }
 
+  private static ResourceManagerException translate(Status status) {
+    int code;
+    // reference: https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+    switch (status.getCode()) {
+      case 0:
+        code = 200;
+        break;
+      case 1:
+        code = 499;
+        break;
+      case 2:
+        code = 500;
+        break;
+      case 3:
+        code = 400;
+        break;
+      case 4:
+        code = 504;
+        break;
+      case 5:
+        code = 404;
+        break;
+      case 6:
+        code = 409;
+        break;
+      case 7:
+        code = 403;
+        break;
+      case 16:
+        code = 401;
+        break;
+      case 8:
+        code = 429;
+        break;
+      case 9:
+        code = 400;
+        break;
+      case 10:
+        code = 409;
+        break;
+      case 11:
+        code = 400;
+        break;
+      case 12:
+        code = 501;
+        break;
+      case 13:
+        code = 500;
+        break;
+      case 14:
+        code = 503;
+        break;
+      case 15:
+        code = 500;
+        break;
+      default:
+        code = BaseHttpServiceException.UNKNOWN_CODE;
+    }
+    return new ResourceManagerException(code, status.getMessage());
+  }
+
   @Override
   public Project create(Project project) {
     try {
-      return resourceManager.projects().create(project).execute();
+      final Operation operation = resourceManager.projects().create(project).execute();
+
+      Operation finishedOp =
+          runWithRetries(
+              new Callable<Operation>() {
+                @Override
+                public Operation call() {
+                  try {
+                    return resourceManager.operations().get(operation.getName()).execute();
+                  } catch (IOException ex) {
+                    throw translate(ex);
+                  } catch (Exception ex) {
+                    throw ex;
+                  }
+                }
+              },
+              CREATE_RETRY_SETTINGS,
+              OPERATION_HANDLER,
+              clock);
+      if (finishedOp.getError() != null) {
+        throw translate(finishedOp.getError());
+      }
+
+      // NOTE(pongad): Operation.getResponse() returns a Map<String, Object>.
+      // `(Project) finishedOp.getResponse()` doesn't work,
+      //   because JSON deserializer in execute() didn't know to create a Project object.
```

Comment:
The comment formatting is weird, is it intentional (unexpected indentations)?

Reply:
It was intentional "paragraph heading". Changed to number instead to make this clear.